interfaces/builtin/x11.go

```diff
@@ -60,14 +120,67 @@ socket AF_NETLINK - NETLINK_KOBJECT_UEVENT
 bind
 `
 
+type x11Interface struct{}
+
+func (iface *x11Interface) Name() string {
+	return "x11"
+}
+
+func (iface *x11Interface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              x11Summary,
+		ImplicitOnClassic:    true,
+		BaseDeclarationSlots: x11BaseDeclarationSlots,
+	}
+}
+
+func (iface *x11Interface) SecCompConnectedPlug(spec *seccomp.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	spec.AddSnippet(x11ConnectedPlugSecComp)
+	return nil
+}
+
+func (iface *x11Interface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	spec.AddSnippet(x11ConnectedPlugAppArmor)
```

Comment:
It is worth noting that x11ConnectedPlugAppArmor has `#include <abstractions/X>` so there won't be a rule in the plugging policy that limits which slot implementation the plug may connect to. On classic distro this is fine but it is also 'ok' in Ubuntu Core since AppArmorConnectedSlot will limit which plugging snaps can connect to it (ie, there are two sides of the connect, so two different rules).

I think you should make a comment in x11ConnectedPlugAppArmor along the lines of:
```
# The X abstraction doesn't check the peer label, but in this case that's
# ok because AppArmorConnectedSlot will limit which clients can connect
# to the slot implementation.
#include <abstractions/X>
```

Reply:
Good point. I've added your comment.